src/openlifu/util/checkgpu.py

```diff
@@ -0,0 +1,12 @@
+from pynvml import nvmlDeviceGetCount, nvmlInit, nvmlShutdown
+
+
+def gpu_available() -> bool:
+    """Check the system for an nvidia gpu and return whether one is available."""
+    try:
+        nvmlInit()
+        device_count = nvmlDeviceGetCount()
+        nvmlShutdown()
+        return device_count > 0
```

Comment:
This is the part to focus on -- the method of checking for GPU availalability here is to check at the device level whether there is an nvidia device.

So this is not checking for example whether the CUDA toolkit is installed.

pynvml is super lightweight of a dependency compared to things that bring in cuda, so that's why I defaulted to that, but please let me know if the relevant thing to check for the purpose of k-wave gpu usage is actually the presence of CUDA

Reply:
yeah, I think this is fine. _Maybe_ we'd want helpful messaging if the CUDA one tries to run and fails, but it could be a pain to test (we'd need a computer with an NVIDIA-but-not-CUDA GPU...)